### PR TITLE
Add RPT transaction page and supporting Java classes

### PR DESCRIPTION
**Pull request by faisal-amin82**
This PR adds code related to RPT transactions which enables users to submit their transactions resource.
It also updates the version of api sdk java to 4.3.5 to include links for rpt transactions api.
Classes include controller, service, transformers and related unit tests and also html page for front end.
This is just a skeleton page with no validations and just the related fields for users to submit relevant information
Resolves: BI-5336

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    @Autowired
+    private ValidationContext validationContext;
+
+    private static final UriTemplate RPT_TRANSACTIONS_URI =
+            new UriTemplate("/transactions/{transactionId}/company-accounts/{companyAccountsId}/small-full/notes/related-party-transactions/transactions");
+
+    private static final UriTemplate RPT_TRANSACTIONS_URI_WITH_ID =
+            new UriTemplate("/transactions/{transactionId}/company-accounts/{companyAccountsId}/small-full/notes/related-party-transactions/transactions/{transactionId}");
+
+    private static final String RESOURCE_NAME = "transactions";
```
Comment (amoriartyCH) [inline src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java:46]: Are they "transactions" or "rptTransactions" ?

File: src/main/resources/templates/smallfull/addOrRemoveRptTransactions.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        <div class="govuk-grid-column-two-thirds">
+
+            <form id="correspondence-form" class="form" th:action="@{''}" method="post">
+                <div th:replace="fragments/numberedHeading :: numberedHeading (
+                  headingText = 'Related party transactions'
+              )">
+                </div>
+
+                <span id="relatedPartyTransactions-help-text" class="govuk-hint">
+                You can report up to 20 related party transactions.
```
Comment (amoriartyCH) [inline src/main/resources/templates/smallfull/addOrRemoveRptTransactions.html:23]: Shouldn't this page have all of the input fields as well as the Add another button but with no functionality surrounding them? As the story on JIRA suggests implementing the page but no functionality.

File: src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    public List<ValidationError> createRptTransaction(String transactionId, String companyAccountsId, RptTransactionToAdd rptTransactionToAdd) throws ServiceException {
+
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        String uri = RPT_TRANSACTIONS_URI.expand(transactionId, companyAccountsId).toString();
+
+        RptTransactionApi rptTransactionApi = rptTransactionsTransformer.getRptTransactionsApi(rptTransactionToAdd);
+
+        try {
+            ApiResponse<RptTransactionApi> apiResponse = apiClient.smallFull().relatedPartyTransactions().rptTransactions().create(uri, rptTransactionApi).execute();
```
Comment (amoriartyCH) [inline src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java:77]: Sonar is complaining about this apiResponse not being used. Should it be returned or used?

File: src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    @Autowired
+    private RptTransactionsTransformer rptTransactionsTransformer;
+
+    @Autowired
+    private ServiceExceptionHandler serviceExceptionHandler;
+
+    @Autowired
+    private ValidationContext validationContext;
+
+    private static final UriTemplate RPT_TRANSACTIONS_URI =
```
Comment (ch-tstrickland) [inline src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java:40]: Move constants to top of class

File: src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java (hunk 1 above)
Comment (ch-tstrickland) [reply src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java:46]: I think that, within the JSON, these are 'transactions', aren't they?

Comment (ch-tstrickland) [reply src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java:46]: It is a little confusing to have it sometimes be transactions, sometimes rptTransactions.

File: src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java (hunk 4 above)
Comment (faisal-amin82) [reply src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java]: Moved constants to the top of the class

File: src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java (hunk 3 above)
Comment (faisal-amin82) [reply src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java]: Removed unused variables

File: src/main/resources/templates/smallfull/addOrRemoveRptTransactions.html (hunk 2 above)
Comment (faisal-amin82) [reply src/main/resources/templates/smallfull/addOrRemoveRptTransactions.html]: Added fields to the page

File: src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java (hunk 1 above)
Comment (faisal-amin82) [reply src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java]: Classes and fields clash in some of the cases so i have used a consistent approach to declare those as RptTransactions while resource name is "transactions" and that is why i have assigned it transactions string value 👍

File: src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/AddOrRemoveRptTransactionsController.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+import uk.gov.companieshouse.web.accounts.service.smallfull.RptTransactionService;
+import uk.gov.companieshouse.web.accounts.service.smallfull.SmallFullService;
+
+import javax.servlet.http.HttpServletRequest;
+
+@Controller
+@NextController(OffBalanceSheetArrangementsQuestionController.class)
+@PreviousController(RelatedPartyTransactionsQuestionController.class)
+@RequestMapping("/company/{companyNumber}/transaction/{transactionId}/company-accounts/{companyAccountsId}/small-full/note/add-or-remove-transactions")
+public class AddOrRemoveRptTransactionsController extends BaseController {
```
Comment (amoriartyCH) [inline src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/AddOrRemoveRptTransactionsController.java:30]: Sorry, just noticed we should be implementing conditional controller here and overwriting the WillRender method

**Review by amoriartyCH — APPROVED** [review]

**Pull request merged**